lib/index.js

```diff
@@ -12,26 +14,42 @@ module.exports = function(server, options, next) {
   var redirector = function(request, reply) {
     var routePath = request.route.path;
     var redirectTo = options.redirects[routePath];
+    var vhost = request.route.settings.vhost;
+    if (vhost)
+      redirectTo = options.vhosts[vhost] ? options.vhosts[vhost][routePath] : undefined
     if(redirectTo) {
       if (log) {
-        server.log(['hapi-redirects'], { routePath: routePath, from: request.path, to: redirectTo });
+        server.log(['hapi-redirects'], { routePath: routePath, from: request.path, to: redirectTo, vhost: vhost });
       }
       return reply.redirect(redirectTo);
     }
-
-    server.log(['hapi-redirects', 'error'], { path: request.path });
+    server.log(['hapi-redirects', 'error'], { path: request.path, vhost : vhost });
     reply(new Error('invalid redirect'));
   };
 
-  for(var route in options.redirects) {
-    routes.push({
-      path: route,
+  function registerRoute(path, vhost){
```

Comment:
lets make sure we are consistent stylistically.  `var registerRoute = function()`